core/src/test/java/com/netflix/iceberg/TestTables.java

```diff
@@ -173,14 +174,39 @@ public void commit(TableMetadata base, TableMetadata metadata) {
       }
     }
 
+    @Override
+    public FileIO io() {
+      return new LocalFileIO();
+    }
+
+    @Override
+    public String metadataFileLocation(String fileName) {
+      return new File(metadata, fileName).getAbsolutePath();
+    }
+
+    @Override
+    public long newSnapshotId() {
+      long nextSnapshotId = lastSnapshotId + 1;
+      this.lastSnapshotId = nextSnapshotId;
+      return nextSnapshotId;
+    }
+  }
+
+  static class LocalFileIO implements FileIO {
+
     @Override
     public InputFile newInputFile(String path) {
       return Files.localInput(path);
     }
 
     @Override
-    public OutputFile newMetadataFile(String filename) {
-      return Files.localOutput(new File(metadata, filename));
+    public OutputFile newOutputFile(String path) {
+      File file = new File(path);
+      if (!file.getParentFile().isDirectory() && !file.getParentFile().mkdirs()) {
```

Comment:
Why does this run `mkdir`? I don't think the output file should make any changes until requested by a `create` call.

Reply:
I see below that one of the tests calls mkdir in newMetadataLocation. I think what I suggested here is correct and that use was wrong.